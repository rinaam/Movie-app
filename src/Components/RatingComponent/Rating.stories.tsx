import React from "react";
import { Story, Meta } from "@storybook/react";

import RatingComponent, { IRatingProps } from "./RatingComponent";

export default {
  title: "Rating",
  component: RatingComponent,
} as Meta;

const Template: Story<IRatingProps> = (args) => <RatingComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  rating: 5.2,
};
