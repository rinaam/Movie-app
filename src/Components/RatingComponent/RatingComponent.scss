@use "sass:map";
@import "../../Styles/breakpoints.scss";
@import "../../Styles/variables.scss";

.rating {
  margin: 24px 0px;
  display: flex;
  &-icon {
    margin-right: map.get($spacing, "xs");
    cursor: pointer;
    color: $yellow;
    font-size: map.get($font-size, "m");
    @include breakpoint(smallDevices) {
      font-size: map.get($font-size, "l");
    }
  }
}
