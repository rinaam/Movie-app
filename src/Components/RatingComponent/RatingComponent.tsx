import React, { useState } from "react";
import "./RatingComponent.scss";
import SolidStar from "../../Assets/Icons/SolidStarIcon.svg";
import RegularStar from "../../Assets/Icons/RegularStarIcon.svg";
import { RATING_GRADES } from "../../Constants";

export interface IRatingProps {
  rating: number;
  onClick: (event: number) => void;
}

const RatingComponent: React.FC<IRatingProps> = ({ rating, onClick }) => {
  const [selectedGrade, setSelectedGrade] = useState(rating);

  const handleClick = (grade: number) => {
    setSelectedGrade(grade);
    onClick(grade);
  };

  return (
    <div className="rating">
      {RATING_GRADES.map((grade) => {
        const StarIcon = grade <= selectedGrade ? SolidStar : RegularStar;
        return (
          <div
            onClick={() => handleClick(grade)}
            className="rating-icon"
            key={grade}
          >
            <StarIcon />
          </div>
        );
      })}
    </div>
  );
};

export default RatingComponent;
