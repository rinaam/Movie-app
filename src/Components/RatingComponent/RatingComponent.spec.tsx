import { act, create } from "react-test-renderer";
import RatingComponent from "./RatingComponent";
import React from "react";

describe("RatingComponent", () => {
  it("should render RatingComponent", () => {
    const fakeClick = jest.fn();
    const wrapper = create(
      <RatingComponent rating={5} onClick={fakeClick} />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle click", () => {
    const fakeClick = jest.fn();
    const wrapper = create(<RatingComponent rating={5} onClick={fakeClick} />);

    const star3 = wrapper.root.findAllByProps({ className: "rating-icon" })[2];
    act(() => {
      star3.props.onClick();
    });

    expect(fakeClick).toBeCalled();
    expect(fakeClick).toBeCalledWith(3);
  });
});
