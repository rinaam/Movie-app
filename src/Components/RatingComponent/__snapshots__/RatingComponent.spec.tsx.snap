// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingComponent should render RatingComponent 1`] = `
<div
  className="rating"
>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
  <div
    className="rating-icon"
    onClick={[Function]}
  >
    <IconMock />
  </div>
</div>
`;
